test/api/plugin-api.spec.js

```diff
@@ -166,5 +166,40 @@ describe("plugin API", () => {
       expect(parser.parse("x", { startRule: "b" })).to.equal("x");
       expect(parser.parse("x", { startRule: "c" })).to.equal("x");
     });
+
+    it("can munge the bytecode", () => {
+      // This test demonstrates that plugins can munge the bytecode, but
+      // is also here to ensure full coverage in generate-js.js.
+      // With the current bytecode generator, compileInputChunkCondition
+      // always ensures that skipAccept will be set on any ACCEPT_N with
+      // a length greater than one. This means that the line that would
+      // normally handle that gets no coverage.
+      //
+      // We fix that here by inserting an effective no-op (PUSH_CUR_POS, POP)
+      // which will prevent compileInputChunkCondition from recognizing the
+      // MATCH_STRING_IC .. ACCEPT_N pattern.
+      const plugin = {
+        use(config) {
+          function munge(ast) {
+            expect(ast.rules.length).to.equal(1);
+            const bc = ast.rules[0].bytecode;
+            expect(bc[0]).to.equal(19 /* MATCH_STRING_IC */);
+            bc.splice(4, 0, 5 /* PUSH_CUR_POS */, 6 /* POP */);
```

Comment:
It is better to use real constants instead of comments. Import opcodes and
```suggestion
            expect(bc[0]).to.equal(op.MATCH_STRING_IC);
            bc.splice(4, 0, op.PUSH_CUR_POS, op.POP);
```